public/utils/alerting.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import rison from 'rison-node';
+import { stringify } from 'query-string';
+import { buildCustomFilter } from '../../../../src/plugins/data/common';
+import { url } from '../../../../src/plugins/opensearch_dashboards_utils/public';
+import {
+  DataPublicPluginStart,
+  opensearchFilters,
+  DuplicateIndexPatternError,
+  IndexPattern,
+} from '../../../../src/plugins/data/public';
+import { CoreStart } from '../../../../src/core/public';
+
+export const buildFilter = (indexPatternId: string, dsl: Record<string, unknown>) => {
+  const filterAlias = 'Alerting-filters';
+  return buildCustomFilter(
+    indexPatternId,
+    dsl,
+    false,
+    false,
+    filterAlias,
+    opensearchFilters.FilterStateStore.APP_STATE
+  );
+};
+
+export const createIndexPatterns = async (
+  dataStart: DataPublicPluginStart,
+  patternName: string,
+  timeFieldName: string,
+  dataSourceId?: string
+) => {
+  let pattern: IndexPattern | undefined;
+  const dataSourceRef = dataSourceId
+    ? {
+        type: 'data-source',
+        id: dataSourceId,
+      }
+    : undefined;
+  try {
+    pattern = await dataStart.indexPatterns.createAndSave({
+      id: '',
+      title: patternName,
+      timeFieldName,
+      dataSourceRef,
+    });
```

Comment:
what does `id: ''` mean here? can we remove it if we are going to use auto generated one

Reply:
No, this is the pattern which is already used for creating new index pattern.

Reply:
got it, thanks for clarification